### ISSUE TITLE
Document how to test for promise rejection

### ISSUE THREAD

**Issue by mk-pmb** [jprichardson/tape-promise#14]
My test shall verify that my async function throws, but `t.throws()` doesn't deal with promise results and `t.rejects` is not a function, so which is it?

**Comment by RyanZim** [comment]
This is currently a bit painful; here's how I'd do it for now:

```js
test('should reject', async (t) => {
  try {
    await shouldReject()
  } catch (e) {
    t.assert(e)
    t.end()
  }
})
```

If the error does not occur, you'll get the error `Test exited without ending`.

@jprichardson Should we consider adding an easier way to deal with this, like `t.rejects`?

**Comment by mk-pmb** [comment]
Then at least I haven't missed something.

:+1: for `.rejects` and `.doesNotReject`. With the current package name and package description, I'd consider their absence a critical bug.

My current workaround is to use [rejected-or-not](https://github.com/twada/rejected-or-not), await its result, then call `t.ok()`. Of course a more specific TAP output would be nicer.¹
I tried add glue code for the failure case but it grew too big and ugly and since my test was the wrong place for it anyway, I discarded it.

It's important to me that `.rejects` and `.doesNotReject` return promises that only resolve when the test criterion is decided. I'm not sure whether they should reject if the criterion is not met. My gut thinks they should always resolve, maybe with a Boolean.

¹ Is there a standard for `.rejects` and `.doesNotReject` TAP result lines? In the long term, we should use that then; for now I'd be happy to use the messages that `.throws` and `.doesNotThrow` would use.

**Comment by jprichardson** [comment]
@RyanZim why not wrap in `aw` and test the presence of an error?

**Comment by RyanZim** [comment]
That's one option that works in most cases.

**Comment by mk-pmb** [comment]
Any progress on this?

**Comment by RyanZim** [comment]
Published in v4.0.0 :tada:
